### DISCUSSION TITLE
Is it possible for other transaction types than withdrawal to have an assigned budget?

### DISCUSSION THREAD

**Discussion by niklas152** [firefly-iii/firefly-iii#6171] [Firefly III]
Hello everyone,

obviously withdrawal can be assigned a budget, from which the transaction amount is then substracted.
However, imagine a budget "Clothing" or the likes, where you first purchase something only to later, for whatever reason, need to return it. Currently, as far as I can see, there is no way to assign a budget to the deposit of the refund which would result in re-adding the amount to the budget.
Or is there a way?

Kind regards,
Niklas

**Comment by JC5** [comment]
https://docs.firefly-iii.org/firefly-iii/faq/financials/#what-do-i-do-with-people-who-pay-me-back

**Comment by skorvek** [comment]
I have a different use case for assigning a transfer to a budget not involving paying someone back:

I have a store credit card (box box store for example) and buy something on the card using one of their promotions- a large appliance on 6 months no interest for example, and then pay it back over time.

The account is set up as an "asset" account per Firefly schema despite it being a liability owed.  It's not a fixed liability like a mortgage- it's a line of credit, I can buy something else any time on that card.

So to pay for that item, I have to perform a transfer from a bank account into the asset account. That transfer cannot be associated with a bill and cannot be associated with the budget. 

Yet- it's a fixed expense for the duration of that liability.  It's money I cannot spend on something else, and it's something for which I need to make regular payments, just like my other bills.  The bill function is perfect- I can set an amount/range, due date and end date.

It's not set up as a liability like a mortgage or car loan. I guess one could set up a separate liability account for the loan, but then that's really confusing as the card balance is the asset balance and the liability balance and hard to keep track of, especially if the asset card is  used for another purpose as well and the month's payment goes partly to the liability and part to regular transfers (and I'm not sure you can even split the transaction that way- would have to test).

The budget part sort of makes sense (although I can see it the other way as well)-  the capital budget was spent on the appliance so the expense itself goes to the budget- but that will end up showing as a lumpy budget report- huge outlier expense outside the capital budget amount then nothing for 6 months while it's paid back. And that's not a budget really. 

The only other alternative I see within the current constraints of the application is to not record the purchase as one large expense, but record each periodic payment as the expense in the asset account, and associate those expense transactions with a bill and budget. Of course, that is not going to match reality and required hand work and maintenance. You won't be able to take advantage of importing transactions (in fact will have to edit imported transactions) your actual balance will not match, and will require you to keep track of the liability amounts and timing manually or enter them as future transactions. Sounds like a headache.

In my setup I did create a "capital budget" to set aside a pot of money each month to make capital improvements to my property, pay for large purchases (appliances, cars), etc. That will help define the amount of money outside of fixed bills and routine expenses to be used for that type of thing. And it's a monthly amount- so when I've paid off that fridge, I have some extra funds each month in that budget to put toward another capital purchase.... or to accelerate payoff of another previous commitment. 

And this is different than a piggy bank. The piggy bank requires cash up front. Good financial discipline to not spend money you don't have, but then again, if I had enough free cash lying around to purchase these items without utilizing credit/free financing, etc- I likely wouldn't need to keep close track of my finances with FireFly.

This seems analogous to another discussion about fixed installment payments from a store. I am putting it here though because the original discussion is about transaction types and budgets (and bills in my case)

**Comment by isaacgr** [comment]
+1 on this. I transfer money out to an RRSP/TFSA. I track that as an asset account, since it adds to my networth and is available (though hopefully never touched), but its a monthly payment that comes out of my chequing that I track. I would want to add this to a budget, since the amount I transfer out can also fluctuate if I choose to increase it or decrease it that month.

**Comment by LuvMug37** [comment]
+1 on this. I share a checking account with my spouse for shared automatic bill payments. This helps us make budgeted deposits for each bill, pay our equal share, and protect our main accounts from third-party access. However, since this account is an asset account in Firefly, I can't assign the appropriate budget to each transfer, making budget tracking a manual hassle (or an approximation for the worse). 

Changing this account to an expense account is not ideal because I want to easily track and compare the balance in Firefly to the actual balance in the banking app. Manually searching for this account in the sea of Firefly expense accounts would be very impractical. 

I would be very grateful for this feature! Thank you
